models/node_options.go

```diff
@@ -40,6 +31,14 @@ type NodeOptions struct {
 
 type NodeOption func(*NodeOptions)
 
+func WithContext(ctx context.Context) NodeOption {
+	return func(n *NodeOptions) {
+		if ctx != nil {
```

Comment:
Since `n.Context` is nil by default, is this check here to prevent overriding an existing non-nil context?

Reply:
nope, thats a copy/paste error from the logger option right below it....removing